pages/instructorGradebook/instructorGradebook.ejs

```diff
@@ -165,62 +165,78 @@
-               data-minimum-count-columns="0"
-               data-search="true"
-               data-show-columns="true"
-               data-show-columns-toggle-all="true"
-               data-show-refresh="true"
-               data-pagination="true"
-               data-pagination-v-align="both"
-               data-pagination-h-align="left"
-               data-pagination-detail-h-align="right"
-               data-page-list="[10,20,50,100,200,500,unlimited]"
-               data-page-size="50"
-               data-smart-display="false"
-               data-show-extended-pagination="true"
-               data-toolbar=".fixed-table-pagination:nth(0)"
-               data-sticky-header="true">
-            <thead>
-              <tr>
-                <th data-field="uid"
-                    data-sortable="true"
-                    data-class="sticky-column text-nowrap"
-                    data-switchable="false">UID</th>
-                <th data-field="uin"
-                    data-sortable="true"
-                    data-class="text-nowrap"
-                    data-switchable="false">UIN</th>
-                <th data-field="user_name"
-                    data-sortable="true"
-                    data-class="text-nowrap"
-                    data-switchable="false">Name</th>
-                <th data-field="role"
-                    data-sortable="true"
-                    data-sort-order="desc"
-                    data-switchable="false">
-                  Role
-                  <button class="btn btn-xs" type="button" title="Show roles help" data-toggle="modal" data-target="#role-help">
-                    <i class="bi-question-circle-fill" aria-hidden="true"></i>
-                  </button>
-                </th>
-                <% course_assessments.forEach(function(assessment, iAssessment) { %>
-                <th data-field="score_<%= assessment.assessment_id %>"
-                    data-sortable="true"
-                    data-class="text-nowrap"
-                    data-formatter="gradeFormatter"
-                    data-sort-order="desc"
-                    data-search-formatter="false"><%- include('../partials/assessment', {assessment: assessment}); %></th>
-                <% }); %>
-              </tr>
-            </thead>
-        </table>
+        <% if (!authz_data.has_course_instance_permission_view) { %>
+          <div class="card-body">
+            <h2>Missing student data access</h2>
+            <p>You must have permission to view student data in order to access the gradebook.</p>
+            <% if (authz_data.has_course_permission_own) { %>
+              <p>
+                You can grant yourself access to student data on the course's
+                <a href="<%= urlPrefix %>/course_admin/staff">Staff tab</a>.
+              </p>
+            <% } else { %>
+              <p>You can request access to the student data by contacting a course owner.</p>
```

Comment:
Would it be reasonable to explicitly list the course owners? We have the problem every semester where someone has inherited a course and comes to us to ask for more permissions because they don't know who the owner is that they need to ask for perms.

Reply:
I think that's totally reasonable, yeah. I'll make that change.

Reply:
If it's painful to obtain the list of owners on every page, we could just unconditionally direct people to the (now globally visible) Staff page, and just show the list of owners there.